File / New / Bnd Osgi Workspace: NoClassDefFoundException javax/net/ssl/SSLContext in GitCloneTemplate.generateOutputs() in Git git = cloneCmd.call();

Eclipse SDK
Version: 2022-09 (4.25)
Build id: I20220831-1800
Java 17.0.9

while working on the templates today I discovered an error when I wanted to create a Bnd Workspace via a workspace template on github.

1. File / New / Bnd Osgi Workspace

<img width="586" alt="image" src="https://github.com/bndtools/bnd/assets/188422/c55255fd-68f1-45e5-821d-2bcb8b638cf3">

2. Choose one of the existing ones (also my own didn't work)

<img width="586" alt="image" src="https://github.com/bndtools/bnd/assets/188422/d86d59f6-a7f5-4206-804a-df8acd427e76">

3. Error on next page

<img width="595" alt="image" src="https://github.com/bndtools/bnd/assets/188422/0f5d3e87-a812-48f5-90cd-3bbd3a56e6a6">


The error **only** happened in the Eclipse instance I started from the Eclipse instance containing the bndtools workspace.

It works **fine** when I do it with my normal Eclipse where I am developing for bndtools (Eclipse 2023-09)

## Debugging

- I happens in `org.bndtools.templating.jgit.GitCloneTemplate.generateOutputs(Map<String, List<Object>>, IProgressMonitor)`


<img width="1728" alt="image" src="https://github.com/bndtools/bnd/assets/188422/c8983dc4-4e67-4a10-8ea9-af32a8f040b9">

<img width="1728" alt="image" src="https://github.com/bndtools/bnd/assets/188422/d6964dfd-7c3e-4bf6-82a5-1699c209d386">


## What did I want to do

I actually wanted to test https://github.com/bndtools/bnd/pull/5907 with the update template in
https://github.com/bndtools/bndtools.workspace.min/pull/11 and https://github.com/bndtools/workspace/pull/24

@pkriens here is one. I needed to modify the code because it is currently swallowed.

<img width="577" alt="image" src="https://github.com/bndtools/bnd/assets/188422/d2f60171-383b-479a-8491-4d17ca130ffe">


```
g! 2023-12-01 14:09:14,238 [HttpClient,https://api.github.com/repos/bndtools/workspace] INFO  a.bnd.build.LoggingProgressPlugin - Download https://api.github.com/repos/bndtools/workspace
2023-12-01 14:09:14,471 [HttpClient,https://api.github.com/repos/bndtools/bndtools.workspace.min] INFO  a.bnd.build.LoggingProgressPlugin - Download https://api.github.com/repos/bndtools/bndtools.workspace.min
java.lang.NoClassDefFoundError: javax/net/ssl/SSLContext
	at org.eclipse.jgit.transport.http.apache.HttpClientConnectionFactory$HttpClientSession.configure(HttpClientConnectionFactory.java:79)
	at org.eclipse.jgit.transport.http.apache.HttpClientConnectionFactory$HttpClientSession.configure(HttpClientConnectionFactory.java:1)
	at org.eclipse.jgit.transport.TransportHttp.httpOpen(TransportHttp.java:1049)
	at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:642)
	at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:465)
	at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:151)
	at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:103)
	at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1405)
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:233)
	at org.eclipse.jgit.api.CloneCommand.fetch(CloneCommand.java:325)
	at org.eclipse.jgit.api.CloneCommand.call(CloneCommand.java:191)
	at org.bndtools.templating.jgit.GitCloneTemplate.generateOutputs(GitCloneTemplate.java:122)
	at bndtools.wizards.workspace.WorkspacePreviewPage.lambda$1(WorkspacePreviewPage.java:144)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)
Caused by: java.lang.ClassNotFoundException: javax.net.ssl.SSLContext cannot be found by org.eclipse.jgit.http.apache_6.3.0.202209071007-r
	at org.eclipse.osgi.internal.loader.BundleLoader.generateException(BundleLoader.java:541)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:536)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:416)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 14 more
```

--- a/bndtools.core/bndtools.shared.bndrun
+++ b/bndtools.core/bndtools.shared.bndrun
@@ -22,7 +22,7 @@
 	eclipse.product=org.eclipse.sdk.ide,\
 	osgi.console=,\
 	osgi.instance.area.default=../bndtools.test/workspace,\
-	org.osgi.framework.bootdelegation='javax.net,javax.management,org.xml.sax,org.xml.sax.helpers,javax.xml,javax.xml.parsers',\
+	org.osgi.framework.bootdelegation='javax.*,org.xml.*',\
 	logback.configurationFile=${fileuri;${.}/logback.xml}
 
 # Keep sorted so that we can diff

